### PR TITLE
feat(sequencer): check for sufficient balance in `check_tx`

### LINKED ISSUE #855: sequencer: add balance check in stateful check for `check_tx`
currently, we don't check sufficient balance for fee payment in the stateful `check_tx` check. however, we want to check this to avoid having excessive invalid txs in the mempool.

### PR DESCRIPTION
**Pull request by noot**
## Summary
check for sufficient balance for all actions/assets in `check_tx`

## Background
don't want the mempool to be filled with invalid txs

## Changes
- check for sufficient balance for all actions/assets in `check_tx`. sums the amount that will be used by each action in the tx by their asset ID and checks for sufficient balance for all asset IDs

## Testing
unit tests

## Related Issues

closes  #855

### CODE REVIEW

File: crates/astria-core/src/sequencer/v1/abci.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ impl AbciErrorCode {
     pub const INTERNAL_ERROR: Self = Self(3);
     pub const INVALID_NONCE: Self = Self(4);
     pub const TRANSACTION_TOO_LARGE: Self = Self(5);
+    pub const INSUFFICIENT_FUNDS: Self = Self(6);
```
Comment (joroshiba) [inline crates/astria-core/src/sequencer/v1/abci.rs:20]: Should there be a matching ABCI error code added here? https://github.com/astriaorg/astria/blob/main/proto/sequencerapis/astria/sequencer/v1/abci.proto#L5

Comment (noot) [reply crates/astria-core/src/sequencer/v1/abci.rs:18]: turns out these are unused, so i removed them

Comment (noot) [reply crates/astria-core/src/sequencer/v1/abci.rs:18]: ugh this is going to break the protos in ci even though these were unused :/ let me know if that's ok

File: crates/astria-sequencer/src/transaction/mod.rs [hunk 2]
```diff
@@ -309,3 +388,121 @@ impl ActionHandler for UnsignedTransaction {
...
+        };
+
+        let signed_tx = tx.into_signed(&alice_signing_key);
+        check_balance_mempool(&signed_tx, &state_tx)
+            .await
+            .expect("sufficient balance for all actions");
+    }
+
+    #[tokio::test]
+    async fn check_balance_mempool_insuffient_other_asset_balance() {
```
Comment (Fraser999) [inline crates/astria-sequencer/src/transaction/mod.rs:463]:
```suggestion
    async fn check_balance_mempool_insufficient_other_asset_balance() {
```

File: crates/astria-sequencer/src/transaction/mod.rs [hunk 3]
```diff
@@ -309,3 +388,121 @@ impl ActionHandler for UnsignedTransaction {
...
+
+        let tx = UnsignedTransaction {
+            nonce: 0,
+            actions,
+        };
+
+        let signed_tx = tx.into_signed(&alice_signing_key);
+        check_balance_mempool(&signed_tx, &state_tx)
+            .await
+            .expect_err("insuffient funds for `other` asset");
```
Comment (Fraser999) [inline crates/astria-sequencer/src/transaction/mod.rs:506]:
```suggestion
            .expect_err("insufficient funds for `other` asset");
```
Should we also be checking that the error contains the expected string to ensure it was really `other`'s ID mentioned as being underfunded.

**Review by Fraser999 — APPROVED** [review]
Just a few non-blocking nitpicks, otherwise LGTM.

**Pull request merged**